Transport.php

```diff
@@ -72,4 +80,73 @@ public static function exists($transport)
     {
         return in_array(strtolower($transport), self::get());
     }
+
+    /**
+     * Register a new wrapper in the collection
+     *
+     * @param  string   $protocol The registered protocol ws, wss, http...
+     * @param  callable $factory  The factory callable
+     * @return void
+     */
+    public static function registerWrapper($protocol, callable $factory)
+    {
+        if( isset(static::$_wrappers[$protocol]) ) {
+            throw new Exception(
+                'The protocol wrapper for %s is already registered!',
+                0,
+                $protocol);
+        }
+
+        static::$_wrappers[$protocol] = $factory;
+
+        return;
+    }
+
+    /**
+     * Share a wrapper factory between different protocols
+     * @param  string $protocol The protocol to used
+     * @param  string $shared   The shared used as factory
+     * @return void
+     */
+    public static function shareWrapper($protocol, $shared)
+    {
+        if( !isset(static::$_wrappers[$shared]) ) {
+            throw new Exception(
+                'Can\'t shared an unknown wrapper %s!',
+                0,
+                $shared);
+        }
+
+        static::$_wrappers[$protocol] = $shared;
+
+        return;
+    }
+
+    /**
+     * Check if a wrapper exists in the collection
+     *
+     * @param  string $protocol The protocol to check
+     * @return callable         The linked factory callable
+     */
+    public static function wrapperExists($protocol)
+    {
+        return true === array_key_exists($protocol, static::$_wrappers);
+    }
+
+    /**
+     * Retrieve the wrapper registered for a protocol
+     * @param  string $protocol The protocol to search
+     * @return callable|null    The factory callable if found, null if not
+     */
+    public static function getWrapper($protocol)
+    {
+        if (false === static::wrapperExists($protocol)) {
+            return null;
+        }
+
+        $value = static::$_wrappers[$protocol];
+        return is_string($value) && isset(static::$_wrappers[$value])?
```

Comment:
We only register callable, so we don't need to check whether it is a string or not. Just `return static::$_wrappers[$protocol];` will be ok I guess.